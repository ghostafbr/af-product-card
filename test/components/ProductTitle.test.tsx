import * as React from 'react';
import renderer, {ReactTestRenderer} from 'react-test-renderer';
import {ProductTitle, ProductCard} from '../../src';
import {product1} from '../data/products';

describe('ProductTitle', () => {

    it('show display the component with the correct personalized title', (): void => {
        const wrapper: ReactTestRenderer = renderer.create(<ProductTitle title="Coffee Mug" className="custom-class"/>);

        expect(wrapper.toJSON()).toMatchSnapshot();
    });

    it('should display the component with the name of the product', (): void => {
        const wrapper: ReactTestRenderer = renderer.create(
            <ProductCard product={product1} initialValues={{count: 4, maxCount: 10}}>
                {() => <ProductTitle/>}
            </ProductCard>
        );

        expect(wrapper.toJSON()).toMatchSnapshot();

    });

});
