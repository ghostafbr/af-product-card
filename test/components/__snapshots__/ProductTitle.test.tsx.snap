// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductTitle should display the component with the name of the product 1`] = `
<div
  className="productCard undefined"
>
  <div
    className="productTitle undefined"
  >
    Coffee Mug - Card
  </div>
</div>
`;

exports[`ProductTitle show display the component with the correct personalized title 1`] = `
<div
  className="productTitle custom-class"
>
  Coffee Mug
</div>
`;
